prody/proteins/interactions.py

```diff
@@ -0,0 +1,2478 @@
+    eq2 = math.sqrt( a2 * a2 + b2 * b2 + c2 * c2) 
+    d = d / (eq1 * eq2) 
+    AngleBetweenPlanes = math.degrees(math.acos(d)) 
+    
+    return AngleBetweenPlanes
+    
+    
+def removeDuplicates(list_of_interactions):
+    ls=[]
+    newList = []
+    for no, i in enumerate(list_of_interactions):
+       i = sorted(list(array(i).astype(str)))
+       if i not in ls:
+           ls.append(i)
+           newList.append(list_of_interactions[no])
+    return newList
+
+
+def selectionByKwargs(list_of_interactions, atoms, **kwargs):
+    """Return interactions based on selection"""
+    
+    if 'selection' in kwargs:
+        if 'selection2' in kwargs:
+            ch1 = kwargs['selection'].split()[-1] 
+            ch2 = kwargs['selection2'].split()[-1] 
+            final = [i for i in list_of_interactions if (i[2] == ch1 and i[5] == ch2) or (i[5] == ch1 and i[2] == ch2)]
+        else:
+            p = atoms.select('same residue as protein within 10 of ('+kwargs['selection']+')')
+            x = p.select(kwargs['selection']).getResnames()
+            y = p.select(kwargs['selection']).getResnums()
+            listOfselection = np.unique(list(map(lambda x, y: x + str(y), x, y)))
+            final = [i for i in list_of_interactions if i[0] in listOfselection or i[3] in listOfselection]
+    else:
+        final = list_of_interactions
+    return final
+
+
+def addHydrogens(pdb, method='openbabel', pH=7.0):    
+    """Function will add hydrogens to the protein and ligand structure using Openbabel [NO11]_
+    or PDBFixer with OpenMM.
+    
+    :arg pdb: PDB file name
+    :type pdb: str
+
+    :arg method: Name of program which will be use to fix protein structure.
+            Two alternative options are available: 'openbabel' and 'pdbfixer'.
+            For either option additional software need to be installed:
+            'openbabel': OpenBabel
+            'pdbfixer': PDBFixer and OpenMM
+            default is 'openbabel'
+    :type method: str
+    
+    :arg pH: pH value applyed only for PDBfixer.
+    :type pH: int, float
+    
+    Instalation of Openbabel:
+    conda install -c conda-forge openbabel    
+
+    Find more information here: https://anaconda.org/conda-forge/openbabel
+                                https://github.com/openmm/pdbfixer
+    Program will create new file in the same directory with 'addH_' prefix.
+    
+    .. [NO11] O'Boyle, N. M., Banck M., James C. A., Morley C., Vandermeersch T., Hutchison G. R. 
+    Open Babel: An open chemical toolbox *Journal of cheminformatics* **2011** 3:1-14. """
+    
+    if method == 'openbabel':
+        try:
+            #import openbabel
+            from openbabel import openbabel
+            obconversion = openbabel.OBConversion()
+            obconversion.SetInFormat("pdb")
+            mol = openbabel.OBMol()
+            obconversion.ReadFile(mol, pdb)
+            mol.AddHydrogens()
+            obconversion.WriteFile(mol, 'addH_'+pdb)
```

Comment:
openbabel also doesn't like cif files

Reply:
> doesn't work to add the prefix to the whole filename:
> 
> ```
> ==============================
> *** Open Babel Error  in ReadMolecule
>   ERROR: not a valid PDB file
> 
> ==============================
> *** Open Babel Error  in WriteFile
>   Cannot write to addH_/home/jkrieger/ScipionUserData/projects/D614G_prody/Runs/000554_ProtImportSetOfAtomStructs/extra/7kec.cif
> ```
> 
> you need to use os.path.split as follows:
> 
> ```
> os.path.join(os.path.split(pdb)[0], "addH_" + os.path.split(pdb)[1])
> ```

I've fixed this now.

Reply:
> openbabel also doesn't like cif files

maybe you could use parsePDB and writePDB to give it a pdb file instead. Otherwise, you should probably add a note about that to the docs and raise a ValueError

Reply:
I fixed it by raising an error